### PR TITLE
feat[android]: change readme page handling for custom package install

### PR DESCRIPTION
**Pull request by safolit**
Addresses #1855

## Previous Behavior ##
Custom keyboard installtion on android shows welcome page before installation.
On other platforms before installation a readme.htm is shown and the welcome.htm is displayed after installation.

## New Behavior ##
On android custom keyboard installation behaves the same as on other platforms.
After selection a Custom package the readme-file is shown
Screenshot_1572851164

After finishing the installation the welcome-file is display
Screenshot_1572851234

Goto keyboard details and show welcome.htm from info link
Screenshot_1572852049

Screenshot_1572852056

## Work Done ##
- add Readme-Page-Helper-Method to FileUtils
-  Change PackageAcitivity to show Readme before starting installation
- Change PackageAcitivity to show Welcome page after finishing installation

## Testing 
- Goto Settings/ Add Keyboard from Device
- Select a downloaded custom keyboard package
- Select a keyboard from the list
- See the readme file from package
- Start keyboard installation
- See welcome page from package
- Press OK to close the welcome page

- Goto Settings/Installed languages
- Select the new language
- Select the new keyboard
- Open Help link
- See welcome page from package

### CODE REVIEW

File: android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/PackageActivity.java [hunk 1]
```diff
@@ -153,17 +155,17 @@ public void onPageFinished(WebView view, String url) {
     });
 
     // Determine if ad-hoc distributed KMP contains welcome.htm (case-insensitive) to display
```
Comment (darcywong00) [inline android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/PackageActivity.java]:
```suggestion
     // Determine if ad-hoc distributed KMP contains readme.htm (case-insensitive) to display
```

File: android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/PackageActivity.java [hunk 2]
```diff
@@ -238,6 +240,28 @@ private void showErrorToast(Context context, String message) {
...
+         break;
+       }
+     }
+     if(!_found)
+       return false;
+
+      final Button installButton = (Button) findViewById(R.id.installButton);
+      final Button cancelButton = (Button) findViewById(R.id.cancelButton);
+      installButton.setVisibility(View.GONE);
+      cancelButton.setText("OK");
```
Comment (darcywong00) [inline android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/PackageActivity.java]: ah, originally the webView had two buttons for "install" and "cancel".
But for displaying welcome.htm we'll only need a single "OK" button.

Can you find a way to update the layout for this 
`/android/KMAPro/kMAPro/src/main/res/layout/activity_package_installer.xml`

so we get the 2 buttons on readme.htm and 1 button for welcome.htm?

Maybe define 2 linearlayouts and change the visibility depending on which .htm file is displayed.

Comment (safolit) [reply android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/PackageActivity.java]: I already hide the install button for the welcome page.

Comment (darcywong00) [reply android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/PackageActivity.java]: > cancelButton.setText("OK");

right, but we probably don't want to maintain a cancel button that has a text of "OK". It leads to questions what happens when you click "OK"?

I'm suggesting in the layout to have 1 linearlayout that has the original 2 buttons. followed by another linearlayout that has a single button ("OK"). Then packageActivity would toggle the visibility of which linear layout gets displayed.

----

Also, you'll want to use the string `label_ok` defined in 
`android\KMEA\app\src\main\res\values\strings.xml`.

e.g. `getString(R.string.label_ok)`

This way (someday) we can localize the app.

File: android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/PackageActivity.java [hunk 3]
```diff
@@ -152,18 +151,18 @@ public void onPageFinished(WebView view, String url) {
       }
     });
 
-    // Determine if ad-hoc distributed KMP contains welcome.htm (case-insensitive) to display
-    FileFilter welcomeFilter = new FileFilter() {
+     // Determine if ad-hoc distributed KMP contains readme.htm (case-insensitive) to display
```
Comment (darcywong00) [inline android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/PackageActivity.java]: This PR is mostly there!

My comment is for the code block in lines 89-92
```
   // Silent installation (skip displaying welcome.htm and user confirmation)
    if (silentInstall) {
      installPackage(context, pkgTarget, pkgId);
      return;
    }
```

1. can you update the comment to include readme.htm and welcome.htm?
2. for `silentInstall`, we don't want to display any of the htm files. Can you update `installPackage()` / `loadWelcomePage()` to handle that?

**Pull request merged**